src/core/server/http/http_server.ts

```diff
@@ -270,6 +277,15 @@ export class HttpServer {
     }
   }
 
+  private setupRequestStateAssignment(config: HttpConfig) {
+    this.server!.ext('onRequest', (request, responseToolkit) => {
+      request.app = {
+        requestId: getRequestId(request, config.requestOpaqueId),
+      } as KibanaRequestState;
```

Comment:
The `KibanaRequestState` type turns out not to be super useful since Hapi's types don't have any generics to support specifying the type of the `request.app` property

Reply:
I *think* this is meant to be 'extended' using `declare` statements. 

```ts
/**
 *  User-extensible type for application specific state.
 */
 /* tslint:disable-next-line:no-empty-interface */
export interface ApplicationState {
}
```

But extending their interface and forcecasting is probably fine.

Second point: maybe we want to preserve the potential existing state instead of erasing it?

```ts
request.app = {
  ...request.app,
   requestId: getRequestId(request, config.requestId),
}
```

this feels like a little more futur-proof if we were to add additional data to `request.app` from somewhere else at some point.